ra_aid/proc/interactive.py

```
@@ -1,90 +1,220 @@
+        bufsize=0,
+        close_fds=True,
+        env=env,
+        preexec_fn=os.setsid  # Create new process group for proper signal handling.
+    )
+    os.close(slave_fd)  # Close slave end in the parent process.
+
+    captured_data = []
+    start_time = time.time()
+    was_terminated = False
+
+    def check_timeout():
```

Ruff (0.8.2)

Missing return type annotation for private function `check_timeout`

Add return type annotation: `bool`

(ANN202)